root/static/scripts/relationship-editor/common/entity.js

```diff
@@ -29,6 +29,8 @@ function getDirection(relationship, source) {
   if (source === entities[1]) {
```

Comment:
Is it ever possible to be neither "forwards" nor "backwards"? If not, we should really be throwing an error rather than returning `undefined` or `null`. This is comment **B**

Reply:
`source` should always be either entity0 or entity1, so I'd probably throw an error here at the end, yes.